src/plots/cartesian/category_mode_defaults.js

```diff
@@ -0,0 +1,39 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+module.exports = function handleCategoryModeDefaults(containerIn, containerOut, coerce) {
+
+    if(containerIn.type === 'category') {
+
+        var validCategories = ['trace', 'category ascending', 'category descending', 'array'];
+
+        var properCategoryList = Array.isArray(containerIn.categorylist) && containerIn.categorylist.length > 0;
+
+        if(validCategories.indexOf(containerIn.categorymode) === -1 && properCategoryList) {
```

Comment:
:+1: 

Coercing while keeping track of whether the user input in `containerIn` is valid was the idea behind [`Lib.coerce2`](https://github.com/plotly/plotly.js/blob/db9fd34251f099907837f4fc8c56e74faa13b9e1/src/lib/coerce.js#L321-L326). 

`Lib.coerce2` is not very well implemented at the moment, and I want to change its API to make more convenient.

So, no action needed here. This logic works great and is well tested.